dev/generate-cuda.py

```diff
@@ -233,7 +232,7 @@ def traverse(node, args={}, forvars=[], declared=[]):  # noqa: B006
             casttype = "uint8_t"
         else:
             casttype = node.func.id
-        code = "({0})({1})".format(
+        code = "({})({})".format(
             casttype, traverse(node.args[0], args, copy.copy(forvars), declared)
         )
```

Comment:
Something like this is not covered to an f-string by pyupgrade - it only does simple strings that are clearly readable as f-strings. Personally, I'd rewrite it this way:

```python
traversed = traverse(node.args[0], args, copy.copy(forvars), declared)
code = f"({casttype})({traversed})"
```

Reply:
I agree, though that's a very case-by-case thing.

Reply:
I think all of them could be written as f-string one way or the other, though (but not in an automated way, yes). In a recent project (Particle, I think), I ended up converting everything - only "format later" cases remain as `.format`, which is the main use for it now. If you assign `something=<long expression>` inside `.format`, then it's always just as readable or more readable to just use normal Python assignment before the f-string.

The call for whether `f(x)` or `x.thing.otherthing` should be pre-assigned or inlined is likely preference, though - probably based on character length + number of brackets required. Pyupgrade just doesn't convert it for you at all if it's not simple.